RELEASE.md

```diff
@@ -1,33 +1,69 @@
-To release a new version of qtpy on PyPI:
+# Release Procedure
 
-* Close Github milestone
+To release a new version of QtPy on PyPI:
 
-* git fetch upstream && git merge upstream/master
 
-* git clean -xfdi
+## Prepare
 
-* Update CHANGELOG.md using `loghub` to generate the list of issues and PRs merged to add at the top of the file
+* Close Github milestone and ensure all issues are resolved/moved
 
-    loghub -m vX.X.X spyder-ide/qtpy
+* `git switch master && git pull upstream master` (assuming `master` is the release branch)
 
-* Update `_version.py` (set release version, remove 'dev0')
+* Check `MANIFEST.in` and `setup.cfg` to ensure they are up to date
 
-* git add . && git commit -m 'Release X.X.X'
 
-* python setup.py sdist
+## Commit
 
-* python setup.py bdist_wheel
+* `pip install --upgrade loghub`
 
-* twine check dist/*
+* Update `CHANGELOG.md` using `loghub` to generate the list of issues and PRs merged to add at the top of the file:
 
-* twine upload dist/*
+  ```bash
+  loghub -m vX.Y.Z spyder-ide/qtpy
+  ```
 
-* git tag -a vX.X.X -m 'Release X.X.X'
+* Update `__version__` in `__init__.py` (set release version, remove `.dev0`)
 
-* Update `_version.py` (add 'dev0' and increment minor)
+* `git commit -am 'Release X.Y.Z'`
 
-* git add . && git commit -m 'Back to work'
 
-* git push
+## Build
 
-* git push --tags
+**Note**: We use `pip` instead of `conda` here even on Conda installs, to ensure we always get the latest upstream versions of the build dependencies.
+
+* `git clean -xfdi`
+
+* `python -m pip install --upgrade-strategy eager --upgrade build pip setuptools twine wheel`
+
+* `python -bb -X dev -W error -m build`
+
+
+## Test
+
+* `python -bb -X dev -W error -m pip install dist/*.whl`
```

Comment:
Does this install the newly created wheel in your environment? If that's the case, why do we want that?

Reply:
Yup. Otherwise, the final release version is never actually tested anywhere before going live to the world, either to see if it can be installed successfully, that the installed environment is consistent (`pip check`) or that tests pass after installing. Even if there are no code changes, there could be a packaging error/typo, something in the packaging stack could have been updated since the last PR (which in fact happened on PR #268 and errored the install), or a user mistake somewhere in the release process. Even if the chances are low, the consequences are very high (especially given how many other packages/applications depend on QtPy) so its worth spending just a few extra seconds per release to do a sanity check that it actually works.

The alternative is creating a branch and submitting the code after line 27 as a PR, which would test it on all platforms in a clean env (and is what I do), but that wouldn't match your current standard workflow and take considerably more time (minutes rather than seconds) for diminishing returns, so I didn't suggest it here.

Reply:
> The alternative is creating a branch and submitting the code after line 27 as a PR

I think this is really unnecessary because after the last PR merged in the repo and before the release, there are zero code changes (except for changing the package version). So we can be certain that our tests are passing.

In all my years as a maintainer, I have released no more than 10 broken packages in hundreds of releases. Also, since this adds more work to maintainers, I'm -1 on it (you'd have to recreate the testing env locally and keep it updated, when it's running perfectly fine on Github).

> Even if the chances are low

I think they are.

> the consequences are very high

No. You can always yank the release on PyPI.

> so its worth spending just a few extra seconds per release to do a sanity check that it actually works.

Seconds if we release frequently. But it could take much more if releases are infrequent.

Reply:
Since this is a change in our release process and it has nothing to do with this PR (whose purpose is to use the new way to create packages), I suggest to remove this section and discuss it in a separate issue. That way we could quickly move forward with this.

Reply:
> Since this is a change in our release process and it has nothing to do with this PR (its purpose is to use the new way to create packages), I suggest to remove this section and discuss it in a separate issue. That way we could quickly move forward with this.

Sure, more than fair. I updated the PR accordingly to remove the ``Test`` section from the Release guide, so this should be ready.